tests/unittests/graphOptzTest.cpp

```diff
@@ -872,3 +872,49 @@ TEST_F(GraphOptz, concatElim) {
   // Check that the concat node is gone.
   EXPECT_EQ(countNodeKind(F_, Kinded::Kind::ConcatNodeKind), 0);
 }
+
+// Check the transformation Concat(Reshape(x) * N) -> Reshape(Concat(x * N)).
+TEST_F(GraphOptz, concatReshapes) {
+  const size_t shape[] = {10, 20};
+  const size_t reshape[] = {10, 1, 20};
+  Type t(ElemKind::FloatTy, shape);
+  std::vector<NodeValue> inputs;
+  for (size_t i = 0; i < 50; i++) {
+    // Create 50 reshape nodes whose input is variable of shape {10,20}.
+    // The reshape nodes will transform their input from shape of {10,20}
+    // to {10, 1, 20}.
+    Node *var = F_->getParent()->createVariable(ElemKind::FloatTy, shape,
+                                                "input" + std::to_string(i));
+    auto *RN = F_->createReshape("reshape" + std::to_string(i), var, reshape);
+    inputs.push_back(RN);
+  }
+  // Create a Concat node that concatenates the 50 reshape nodes.
+  auto *CN = F_->createConcat("concat", inputs, 1);
+  auto outputShape = CN->getResult().dims();
+  inputs.erase(inputs.begin(), inputs.end());
+  auto *O = F_->createSave("ret", CN);
+
+  // Before optimization, we have 52 nodes in the graph.
+  EXPECT_EQ(F_->getNodes().size(), 52);
+
+  ::glow::optimize(F_, CompilationMode::Infer);
+
+  // After optimization, we expect to see only 3 nodes.
+  EXPECT_EQ(F_->getNodes().size(), 3);
+
+  // CN should not exist any more.
+  EXPECT_TRUE(std::find_if(F_->getNodes().begin(), F_->getNodes().end(),
```

Comment:
None of the RNs stored in inputs should exist either, right? May be add a check for that.

Reply:
The RNs vector were dereferenced, i cant perform the check tho.